manifests/service.pp

```diff
@@ -0,0 +1,62 @@
+# == Class: xylem::service
+#
+# Installs Xylem and configures and manages the service.
+#
+# === Parameters
+#
+# [*repo_manage*]
+#   If true, xylem::repo will be used to manage the package repository.
+#
+# [*repo_source*]
+#   Repository source passed to xylem::repo.
+#
+# [*package_ensure*]
+#   The ensure value for the seed-xylem package.
+#
+# [*backend*]
+#   Set the rhumba backend to use for job queue
+#
+# [*redis_host*]
+#   Redis host to use when using the redis Rhumba backend
+#
+# [*redis_port*]
+#   Redis port to use when using the redis Rhumba backend
+#
+class xylem::service (
+  $backend           = undef,
+  $redis_host        = undef,
+  $redis_port        = undef,
+
+  $repo_manage       = true,
+  $repo_source       = 'p16n-seed',
+  $package_ensure    = 'installed',
+){
+  validate_bool($repo_manage)
+
+  class { 'xylem::config':
+    backend    => $backend,
+    redis_host => $redis_host,
+    redis_port => $redis_port,
+  }
+
+  if $repo_manage {
+    class { 'xylem::repo':
+      manage => $repo_manage,
+      source => $repo_source,
+    }
+  }
+
+  package { 'seed-xylem':
```

Comment:
The general idea is to have a single class that sets up the basic xylem machinery. That used to be `xylem::node`, but requiring a single definition that includes all the plugins is a problem when you have different kinds of plugins that might coincidentally share a host.

I've called this thing `xylem::service` because I couldn't think of a better name. Do you have any suggestions?